### PR TITLE
Added a problem on Merge K Sorted Arrays

### LINKED ISSUE #1715: <CUSTOM><ADD A PROBLEM> Merge K Sorted Arrays to Top DSA Interview Questions
### Feature Name

 Merge K Sorted Arrays 

### Feature Description

 Merge K Sorted Arrays so that final array is sorted in ascending order.

### Motivation

This is an intuitive algorithm best for interview preparation where person can learn usage of Heaps more efficiently.

### Implementation Suggestions (Optional)

1. Create a min Heap and insert the first element of all the K arrays.
2. Run a loop until the size of Min Heap is greater than zero.
3. Remove the top element of the Min Heap and print the element.
4. Now insert the next element from the same array in which the removed element belonged only if array size greater than zero. 
5. Simply perform these operations until heap size greater than zero.
6. Return the output array.

### Feature Type

New Algorithm

### Does this feature require additional resources?

- [ ] Additional Documentation
- [ ] Third-Party Libraries
- [ ] UI/UX Design
- [ ] Other

### References (Optional)

_No response_

### PR DESCRIPTION
**Pull request by Riddhi12349**
### Description
Merge K Sorted Arrays so that final array is sorted in ascending order.

Fixes #1715 

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/DSA-Problem-Solution/Merge_K_Sorted_Arrays.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
+---
+id: Merge K Sorted Arrays
```
Comment (ajay-dhangar) [inline docs/DSA-Problem-Solution/Merge_K_Sorted_Arrays.md]: replace `id: Merge K Sorted Arrays` with `id: merge-k-sorted-arrays`

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update your code based on suggestions

**Comment by Riddhi12349** [comment]
done

File: docs/DSA-Problem-Solution/Merge_K_Sorted_Arrays.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+## Description
+
+The **Merge K Sorted Arrays** problem is an intuitive problem based on Priority Queue where we have to merge k sorted arrays into one final sorted array.
+
+### Problem Definition
+
+Given:
+
+- A 2D array of integers `arr` of size k\*k.
```
Comment (ajay-dhangar) [inline docs/DSA-Problem-Solution/Merge_K_Sorted_Arrays.md]: replace `k\*k` with `$k\*k$`

File: docs/DSA-Problem-Solution/Merge_K_Sorted_Arrays.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+- Creating a `MinHeap` and Insert the `first` element `of all the k arrays`.
+- Remove the `top most` element of Minheap and `put` it in the output array.
+- And insert the `next` element `from`the `array of removed` elements.
+- To get the `result` the step must continue until there is no element left in the MinHeap.
+
+2. **Return** `result`, which is the final sorted array after merging k sorted arrays.
+
+### Time Complexity
+
+- **Time Complexity**: O(K^2\* log(K)), where insertion and deletion in a Min Heap requires log K time and for all K^2 elements it takes (K^2 \* log(K)) time
```
Comment (ajay-dhangar) [inline docs/DSA-Problem-Solution/Merge_K_Sorted_Arrays.md]: replace `O(K^2\* log(K))` with `$O(K^2\* log(K))$`, `k^2` with `$k^2$`, and `(K^2 \* log(K))` with `$(K^2 \* log(K))$`

File: docs/DSA-Problem-Solution/Merge_K_Sorted_Arrays.md [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+- **Space Complexity**: O(K) for the result array.
```
Comment (ajay-dhangar) [inline docs/DSA-Problem-Solution/Merge_K_Sorted_Arrays.md]: replace `O(K)` with `$O(K)$`

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
Update your code based on suggestions

**Comment by Riddhi12349** [comment]
done

**Review by ajay-dhangar — APPROVED** [review]
Done

**Pull request merged**